src/charts/CytoViz/CytoViz.js

```diff
@@ -84,6 +130,23 @@ export const CytoViz = (props) => {
   const changeZoomPrecision = (event, newValue) => {
     setZoomPrecision(newValue);
   };
+  const toggleAllNodesCollapsed = (event) => {
+    if (compoundsApiRef.current && cytoRef.current) {
+      if (!allCompoundsAreCollapsed) {
+        compoundsApiRef.current.collapseAll(getCompoundApiOptions(currentLayout, useCompactMode, spacingFactor));
+        compoundsApiRef.current.collapseAllEdges();
+        setAllCompoundsAreCollapsed(true);
+      } else {
+        compoundsApiRef.current.expandAllEdges();
+        compoundsApiRef.current.expandAll(getCompoundApiOptions(currentLayout, useCompactMode, spacingFactor));
+        setAllCompoundsAreCollapsed(false);
```

Comment:
Minor suggestions to improve this part :smile: 

- to improve readability you could start without the negation in the `if` condition, if it is followed by an `else` (unless the first block is much shorter than the second one, for instance)
- the result of `getCompoundApiOptions(currentLayout, useCompactMode, spacingFactor)` could be stored in a `compoundApiOptions` variable just before `if (!allCompoundsAreCollapsed)` to prevent repeating twice the function with all its parameters (the code will be easier to maintain if the function signature changes, for example)
- the two calls to `setAllCompoundsAreCollapsed(...)` could be replaced by a single call outside of the if/else block by calling it like this: `setAllCompoundsAreCollapsed(!allCompoundsAreCollapsed);` (`setFoo(!foo)` is a natural pattern for a toggle function :slightly_smiling_face: )